release.md

```diff
@@ -1,41 +1,190 @@
+public facing apis and interfaces.
+* After MAJOR becomes 1, updates that break the public facing api
+or interface need to update MAJOR version.
 * MINOR updates have to be backwards compatible (after MAJOR>=1).
 * PATCH updates are for bug fixes.
 
+Versions based on unreleased branches of master will be suffixed with ".dev".
+
 ## Releases
 
 We use github's release system for creating releases.  Release are listed
 [on the Cirq release page](https://github.com/quantumlib/Cirq/releases).
 
-Our development process uses the `master` branch for development. 
-When a release is made for a major or minor version update, `master`
-is tagged with a version tag (vX.X.X) for a pull request corresponding 
-to the release.  In the pull request corresponding to the release 
-the [version file](cirq/_version.py) should be updated. After
-that version is cut, a future pull request should update the 
-version to the next minor version with `.dev` appended.
-
-For patch version updates (bug fixes), we follow a different pattern.
-For these we create a separate branch that off of the version the
-major minor version for the patch, or a previous branch patch.  The
-branches  should be of the name `branch-X-X-X` corresponding to the 
-version.  These can then be appropriately tagged and the release
-pushed to pypi.  These fixes, if possible, should also be merged
-into master via a separate change.
+Our development process uses the `master` branch for development.
+Master will always use the next unreleased minor version with the suffix
+of ".dev".  When a release is performed, the ".dev" will be removed and tagged
+in a release branch with a version tag (vX.X.X).  Then, master will be updated
+to the next minor version.  This can always be found in the
+[version file](cirq/_version.py).
+
+
+## Release Procedure
+
+This procedure can be followed by authorized cirq developers to perform a
+release.
+
+### Preparation
+
+Make sure you're on an up-to-date master branch and in cirq's root directory.
+
+```bash
+git checkout master
+git pull origin master  # or upstream master
+git status  # should be no pending changes
+```
+
+Ensure you have pypi and test pypi accounts with access to cirq distribution.
+This can be done by visiting test.pypi.org, logging in, and accessing the cirq
+distribution.
+
+For the following script to work, you will need the following env variables
+defined: TEST_TWINE_USERNAME, TEST_TWINE_PASSWORD, PROD_TWINE_USERNAME,
+PROD_TWINE_PASSWORD.
+
+It is highly recommended to use different passwords for test and prod to avoid
+accidentally pushing to prod.
+
+Also define these variables for the versions you are releasing:
+
+```bash
+VER=VERSION_YOU_WANT_TO_RELEASE  # e.g. "0.7.0"
+NEXT_VER=NEXT_VERSION  # e.g. "0.8.0"
+```
+
+### Create release branch
+
+Create a release branch called "v${VERSION}-dev":
+
+```bash
+git checkout master -b "v${VER}-dev"
+vi _version.py   # Remove .dev from version
```

Comment:
or emacs ;)

Reply:
Using emacs would be incorrect, just ask the internet for a uniformly consistent opinion.